File: springboot_flutter/src/main/java/com/example/springboot_flutter/dto/AuthResponse.java
package com.example.springboot_flutter.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthResponse {

    private String token;
    private String refreshToken;

    @Builder.Default
    private String type = "Bearer";

    private Long usuarioId;
    private String email;
    private String nombre;
    private String mensaje;
}

File: flutter_client/lib/models/auth_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'auth_response.g.dart';

@JsonSerializable()
class AuthResponse {
  final String token;
  final String refreshToken;
  final String type;
  final int usuarioId;
  final String email;
  final String nombre;
  final String mensaje;

  AuthResponse({
    required this.token,
    required this.refreshToken,
    required this.type,
    required this.usuarioId,
    required this.email,
    required this.nombre,
    required this.mensaje,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) =>
      _$AuthResponseFromJson(json);

  Map<String, dynamic> toJson() => _$AuthResponseToJson(this);
}